nuxeo-features/nuxeo-platform-rendition/nuxeo-platform-rendition-core/src/main/java/org/nuxeo/ecm/platform/rendition/operation/GetContainerRendition.java

```diff
@@ -0,0 +1,173 @@
+    protected int maxDepth = 5;
+
+    @Param(name = "_currentDepth", required = false)
+    protected int currentDepth = 0;
+
+    @Context
+    protected CoreSession session;
+
+    protected int added = 0;
+
+    protected BlobList getCollectionBlobs(Collection collection) {
+        BlobList blobs = new BlobList();
+        for (String memberId : collection.getCollectedDocumentIds()) {
+            DocumentRef memberRef = new IdRef(memberId);
+            if (session.exists(memberRef)
+                    && !LifeCycleConstants.DELETED_STATE.equals(session.getCurrentLifeCycleState(memberRef))) {
+                DocumentModel member = session.getDocument(memberRef);
+                Blob blob = getDefaultRendition(member);
+                if (blob != null) {
+                    blobs.add(blob);
+                    if (limit > -1 && ++added >= limit) {
+                        break;
+                    }
+                }
+            }
+        }
+        return blobs;
+    }
+
+    protected Blob getDefaultRendition(DocumentModel doc) {
+        Blob blob = null;
+        if ((collectionManager.isCollection(doc) || doc.hasFacet(FacetNames.FOLDERISH)) && currentDepth < maxDepth) {
+            blob = processContainer(doc);
+        } else {
+            Rendition rendition = renditionService.getDefaultRendition(doc, reason, null);
+            if (rendition != null) {
+                blob = rendition.getBlob();
+                if (blob == null) {
+                    log.debug(String.format("Default rendition '%s' has an null Blob for document '%s'",
+                            rendition.getName(), doc.getPathAsString()));
+                }
+            }
+        }
+        return blob;
+    }
+
+    private BlobList getFolderishBlobs(DocumentModel parent) {
+        BlobList blobs = new BlobList();
+        DocumentModelIterator it = session.getChildrenIterator(parent.getRef());
+        while (it.hasNext()) {
+            DocumentModel child = it.next();
+            if (!LifeCycleConstants.DELETED_STATE.equals(session.getCurrentLifeCycleState(child.getRef()))) {
+                Blob blob = getDefaultRendition(child);
+                if (blob != null) {
+                    blobs.add(blob);
+                    if (limit > -1 && ++added >= limit) {
+                        break;
+                    }
+                }
+            }
+        }
+        return blobs;
+    }
+
+    protected Blob processContainer(DocumentModel doc) {
+        AutomationService as = Framework.getLocalService(AutomationService.class);
+        try (OperationContext oc = new OperationContext(session)) {
+            oc.push("limit", limit);
+            oc.push("maxDepth", maxDepth);
+            oc.push("_currentDepth", currentDepth + 1);
+            oc.push("reason", reason);
+            oc.setInput(doc);
+            return (Blob) as.run(oc, ID);
```

Comment:
Why not recurse directly? You don't need to go through the whole Automation machinery. And it would avoid the `_currentDepth` "hack"